nhaystack-rt/src/nhaystack/util/TypeUtil.java

```diff
@@ -226,6 +228,46 @@ else if (def instanceof BSimple)
 
             return simple;
         }
+        else if (def instanceof BEnumOverride)
+        {
+            BEnumOverride cpx = (BEnumOverride) def;
+            BFacets facets = comp.getAction(action.getName()).getFacets();
+            BEnumRange range = (BEnumRange) facets.get(BFacets.RANGE);
+
+            Iterator<Map.Entry<String, HVal>> it = args.iterator();
+            while (it.hasNext())
+            {
+                Map.Entry<String, HVal> e = it.next();
+                BSimple value = toBajaSimple(e.getValue());
+                if (!e.getKey().equals("value"))

```

Comment:
In a BEnumOverride there are only 3 possible fields you can set, duration, value, and maxOverrideDuration.  In this 'if' block what would happen if another key not equals to 'value' and also not one of the 3 permitted fields was passed in?

Reply:
That's true, it is assumed that the user provides only valid inputs. I can rewrite the code to only allow for 3 permitted fields